app/backend/lib/s3client.ts

```diff
@@ -25,3 +31,18 @@ aws.config.update({
 const s3Client = new aws.S3();
 
 export default s3Client;
+
+const awsConfig = {
+  region: AWS_S3_REGION,
+  credentials: fromTemporaryCredentials({
+    masterCredentials: fromEnv(),
+    params: {
+      RoleArn: AWS_ROLE_ARN,
+      RoleSessionName: `s3-v3-role-session-${Date.now()}`,
+      DurationSeconds: 3600,
```

Comment:
we confirmed that when the temporary credentials expire this factory is documented to use the master credentials to refresh the temporary